src/ModelContextProtocol/Server/McpServerResource.cs

```diff
@@ -0,0 +1,17 @@
+using ModelContextProtocol.Protocol.Types;
+
+namespace ModelContextProtocol.Server;
+
+/// <summary>
+/// Represents a resource that the server supports.
+/// </summary>
+public class McpServerResource : IMcpServerPrimitive
+{
+    /// <summary>
+    /// The resource instance.
+    /// </summary>
+    public required Resource ProtocolResource { get; init; }
+
+    /// <inheritdoc />
+    public string Name => ProtocolResource.Name;
+}
```

Comment:
This should have members for being invoked and returning the relevant data, as is the case for McpServerPrompt/Tool.